### ISSUE TITLE
v4.0.0 missing TypeScript declaration files

### ISSUE THREAD

**Issue by bombillazo** [TanStack/query#3846]
### Describe the bug

Version 4.0.0 is missing the declaration files to make use of TypeScript typing.

This is affecting all packages part of v4.0.0:
```
@tanstack/query-core
@tanstack/query-async-storage-persister
@tanstack/react-query-persist-client
etc... 
```

### Your minimal, reproducible example

N/A

### Steps to reproduce

1. Run `yarn add @tanstack/react-query`
2. Import library in component.
3. Get a warning on missing declaration files:
![image](https://user-images.githubusercontent.com/44002321/179614285-8d9c5742-8737-40de-abc9-83caf8137ca8.png)

### Expected behavior

Declaration files are included in the package.

### How often does this bug happen?

Always

### Screenshots or Videos

![image](https://user-images.githubusercontent.com/44002321/179615470-3228c682-9d74-429a-b0cb-59d428713fde.png)

### Platform

N/A

### react-query version

v4.0.0

### TypeScript version

v4.3.5

**Comment by kkirsche** [comment]
Closing https://github.com/TanStack/query/issues/3847 as a dupe of this

**Comment by bombillazo** [comment]
Release [v4.0.2](https://github.com/TanStack/query/releases/tag/v4.0.2) fixes this, but some imports not are not compatible with the old structure. We cannot import things that now live in `@tanstack\react-query` using `@tanstack\react-query`:

```
// before
import {
  useQuery,
  useMutation,
  UseQueryOptions,
  UseMutationOptions,
} from '@tanstack/react-query';

// now must be this to work
import {
  useQuery,
  useMutation,
  UseQueryOptions,
  UseMutationOptions,
} from '@tanstack/react-query/build/types/packages/react-query/src';

```

**Comment by bombillazo** [comment]
Another example:
```ts
// before 
import {
  QueryClient,
  QueryClientProvider,
  QueryFunction,
  QueryFunctionContext,
  QueryKey,
} from 'react-query';

// now must be this to work
import {
  QueryClient,
  QueryFunction,
  QueryFunctionContext,
  QueryKey,
} from '@tanstack/react-query';
import { QueryClientProvider } from '@tanstack/react-query/build/types/packages/react-query/src';
```

**Comment by Karol-Perec** [comment]
#3850 should fix it

**Comment by TkDodo** [comment]
fixed with 4.0.3

**Comment by maapteh** [comment]
@TkDodo please re-open

`Module '"@tanstack/react-query"' has no exported member 'QueryClient'.ts(2305)`
